### PR TITLE
Allow Pipeline configs to be deployed as an official Helm chart.

### PR DESCRIPTION
**Pull request by adjavaherian**
# Changes

This bootstraps the configs with the necessary template, chart files that Helm will look for when attempting to validate and deploy a version of the project configs. ie release.yaml.  It's not made to replace release.yaml but rather give users the ability to override default values and have a quick and easy to reproduce way of deploying the configs as a single chart.

The only caveat here is that all the configs should be moved from the top-level configs directory to a single sub-directory of the charts top level directory.  This will allow the project to support multiple charts, as the charts are required to live in the same directory as the chart name.

Other than that, it looks like there are no automated processes for creating the release yaml that is deployed to Google Storage API.  The bonus side-effect is that helm template can now be used to create the release yaml without additional scripting.

[https://github.com/tektoncd/pipeline/issues/1542](https://github.com/tektoncd/pipeline/issues/1542)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ -] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

Allows users to use Helm to deploy configs directly from GitHub (or Chart repos in future).
eg.
```
- $ helm install tekton-pipelines/tekton-pipelines --name <my-release> [--tls]

```

### CODE REVIEW

**Comment by adjavaherian** [comment]
@googlebot I signed it!

File: charts/index.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  tekton-pipelines:
+  - apiVersion: v1
+    appVersion: "1.0"
+    created: "2019-12-13T16:30:34.228777-05:00"
+    description: A Helm chart for Tekton Pipelines
+    digest: 8f93704f1a1f3befdb454c50ed74d4466b7534f360382b73bc8708b920eb2c63
+    icon: https://raw.githubusercontent.com/jenkins-x/jenkins-x-platform/master/images/go.png
+    name: tekton-pipelines
+    urls:
+    - https://raw.githubusercontent.com/adjavaherian/pipeline/master/charts/tekton-pipelines/tekton-pipelines-0.9.0.tgz
```
Comment (adjavaherian) [inline charts/index.yaml]: This is currently pointing at my personal fork for testing the Helm install.  Will change this url for the official Tekton release.

**Comment by ImJasonH** [comment]
Would it be possible to add this to the tektoncd/experimental repo initially?

That's where we put things we haven't yet committed full support to, which seems to encompass Helm deployment support.

**Comment by sbwsg** [comment]
@adjavaherian Thanks for the PR!  We'd like to encourage you to contribute this to the [experimental repo](https://github.com/tektoncd/experimental) since this isn't something the pipeline project has committed full support to yet.  It would also be great if you [joined us at one of our working groups](https://github.com/tektoncd/community/blob/master/working-groups.md#general) and present your work on this since I'm sure many in the community would like to hear about it!

I'm going to close this PR but strongly encourage you to make a PR [against experimental instead](https://github.com/tektoncd/experimental).  Thanks again!

**Pull request closed without merging**